### ISSUE TITLE
error

### ISSUE THREAD

**Issue by jaccen** [42yeah/GSRast#1]
Bad reader to path: shaders/pointcloud/vertex.glsl?
Bad reader to path: shaders/pointcloud/fragment.glsl?
Invalid shaders supplied?
Bad reader to path: shaders/splats/vertex.glsl?
Bad reader to path: shaders/splats/fragment.glsl?
Invalid shaders supplied?
Bad PLY reader: data.ply?

**Comment by 42yeah** [comment]
Hi @jaccen , you should be running gsrast or any other program in the build directory (so `cd ./build && ./apps/gsrast/gsrast` or your build dir as cwd). Also you need to provide the trained data.ply yourself; if you don't have one yet, I have trained the train scene and you can download it [here](https://files.42yeah.is/api/uploads/WDGyLCkn3RZA6S5XnO8Fsp8ZS2qxAbIG/data.ply).

**Comment by jaccen** [comment]
thanks，get work。
only two model ：points and eliipsoids。 how can i render the normal pic use this project？

**Comment by 42yeah** [comment]
> thanks，get work。 only two model ：points and eliipsoids。 how can i render the normal pic use this project？

True Gaussian rasterization is CUDA-only and you can find the implementation in the [experimental-1](https://github.com/42yeah/GSRast/tree/experimental-1) branch. Related CUDA render engine source code is in `apps/gsrast/gscuda`.

**Comment by jaccen** [comment]
- no plans to implement a 3d gaussian demo that doesn't require CUDA?

**Comment by 42yeah** [comment]
> * no plans to implement a 3d gaussian demo that doesn't require CUDA?

Sorry, not for now. There's no good non-CUDA solution for per-frame depth-based radix sort yet. If there were, maybe I'll try to implement it.

**Comment by jaccen** [comment]
yes  there hava，like  https://github.com/mkkellogg/GaussianSplats3D

**Comment by 42yeah** [comment]
Wow, OK. I did not know that. Thanks for letting me know! Though, from what I've gathered in its README, the splat sorting is on the CPU? I'll look into it; I was planning to implement a CUDA-less version from the very beginning (but that's going to take a while).

**Comment by jaccen** [comment]
yes，i also want to implement a no -cuda with opengl c++ version，i have implented a version from this，https://github.com/mkkellogg/GaussianSplats3D, but  i think  there will be some optimizations should to do，like  computer
 cov3d in shader .  maybe i will set the source on github  later.
